### PR TITLE
Set-cookie header not applied on internal redirect

### PR DESCRIPTION
**Pull request by pmi**

### CODE REVIEW

File: src/main/resources/lib/nextjs-proxy/proxy.es6 [hunk 1]
```diff
@@ -39,14 +40,87 @@ const errorResponse = function (url, status, message, req, renderSingleComponent
...
+    }
+}
+
+function doRequest(req, renderSingleComponent) {
+    const frontendUrl = req.url;
+    const xpSiteUrl = req.headers['xpBaseUrl'];
+
+    const response = httpClientLib.request(req);
+
+    if (Math.floor(response.status / 100) === 3) {
```
Comment (alansemenov) [inline src/main/resources/lib/nextjs-proxy/proxy.es6:74]: `String(response.status).startsWith('3')`?

File: src/main/resources/lib/nextjs-proxy/parsing.es6 [hunk 2]
```diff
@@ -141,18 +141,24 @@ export const parseFrontendRequestPath = (req) => {
 
 
 export const relayUriParams = (req, frontendRequestPath) => {
-    const frontendServerUrl = getFrontendServerUrl();
-    const reqPath = frontendRequestPath?.length ? frontendRequestPath.replace(removeStartSlashPattern, '') : '';
+    let frontendServerUrl = getFrontendServerUrl();
+    let reqPath = frontendRequestPath?.length ? frontendRequestPath.replace(removeStartSlashPattern, '') : '';
```
Comment (poi33) [inline src/main/resources/lib/nextjs-proxy/parsing.es6:145]: `const frontendServerUrl`
`const reqPath` 
?

File: src/main/resources/lib/nextjs-proxy/proxy.es6 [hunk 3]
```diff
@@ -39,14 +40,87 @@ const errorResponse = function (url, status, message, req, renderSingleComponent
...
+    }
+
+    if (isHtml || isJs || isCss) {
+        response.body = getBodyWithReplacedUrls(req, response.body, xpSiteUrlWithoutEditMode);
     }
+
+    response.postProcess = isHtml
+
+
+    log.info(`<--- [${response.status}]: ${frontendUrl}
```
Comment (poi33) [inline src/main/resources/lib/nextjs-proxy/proxy.es6:117]: I don't know if we would want every single component rendered to be in the log?
Could we do this on the debug level?

**Review by poi33 — APPROVED** [review]
Optional changes looks good

**Review by alansemenov — CHANGES_REQUESTED** [review]
Implement the changes mentioned in the comments

File: src/main/resources/lib/nextjs-proxy/proxy.es6 (hunk 1 above)
Comment (pmi) [reply src/main/resources/lib/nextjs-proxy/proxy.es6:74]: Humanitarian vs mathematician ways 😆

Comment (pmi) [reply src/main/resources/lib/nextjs-proxy/proxy.es6:74]: `response.status >= 300 && response.status < 400` is probably the easiest to understand though

File: src/main/resources/lib/nextjs-proxy/proxy.es6 (hunk 3 above)
Comment (pmi) [reply src/main/resources/lib/nextjs-proxy/proxy.es6:117]: It is logged for every request, not every component, but we can change it to debug indeed

File: src/main/resources/lib/nextjs-proxy/parsing.es6 (hunk 2 above)
Comment (pmi) [reply src/main/resources/lib/nextjs-proxy/parsing.es6:145]: reqPath is reassigned later, but frontentServerUrl should be const

**Review by poi33 — APPROVED** [review]

**Pull request merged**